### PR TITLE
Apim 1804 refactor plan security type 

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-1804

## Description

Move plan security type selection outside of api-plan-form component.

PR in 4 commits for readability... Will squash before merging :P 

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-wdhnxtdgzp.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/entities/management-api-v2/plan/planSecurityType.ts [hunk 1]
```diff
@@ -18,3 +18,30 @@
  * Plan security type.
  */
 export type PlanSecurityType = 'KEY_LESS' | 'API_KEY' | 'OAUTH2' | 'JWT' | 'SUBSCRIPTION';
+
+export interface PlanSecurityVM {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/entities/management-api-v2/plan/planSecurityType.ts]: as its not a management-api-v2 type maybe we can keep it in root entities folder ?

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/entities/management-api-v2/plan/planSecurityType.ts]: or next to the service 🤔

**Review by ThibaudAV — COMMENTED** [review]
It seemed to me that these portal settings were loaded once in the application and saved in the global Constant.  
what do you think if we use it to avoid the call every time?

File: gravitee-apim-console-webui/src/entities/management-api-v2/plan/planSecurityType.ts (hunk 1 above)
Comment (jourdiw) [reply gravitee-apim-console-webui/src/entities/management-api-v2/plan/planSecurityType.ts]: Moved to `ConstantsService`

**Review by ThibaudAV — APPROVED** [review]

**Review by loriepisicchio — APPROVED** [review]

**Pull request merged**